virtual-display-driver-cli/src/client.rs

```diff
@@ -1,110 +1,139 @@
-use std::io::Write as _;
+use std::{
+    collections::{BTreeSet, HashSet},
+    io::Write as _,
+};
 
 use driver_ipc::Monitor;
-use eyre::Context;
+use eyre::Context as _;
+use joinery::Joinable as _;
 use win_pipes::{NamedPipeClientReader, NamedPipeClientWriter};
 
 pub struct Client {
-    reader: NamedPipeClientReader,
     writer: NamedPipeClientWriter,
+    state: Vec<Monitor>,
 }
```

Comment:
@MolotovCherry I remember you mentioned that you thought that `Client` might not be the best name now that this type (at least conceptually) manages the driver's state internally. I still feel that client is still a sensible name (it's just a client that's a bit smarter and effectively "caches" some requests), but I'm open to changing this if you have something better in mind. A few possible alternatives I could think of:

- `VirtualDisplayDriver`
- `State`
- `AppState`
- `Cli`

Reply:
Oh, and one interesting consequence of the refactoring of this PR is that the `reader` is now only used on connection. I think this makes sense based on how the IPC messaging works with the driver.

Reply:
I can't think of a better and more descriptive name than `Client`.

Strictly speaking, a way to combat this would be to separate out the writer/reader (though no reader needs to exist now, so only writer) into another struct, name that `Client`, then make it part of this struct with the state, and name it something sensible.

But that's too much work right now, and with very little benefit since everything is small enough.

So, it's fine the way it is.